### PR TITLE
add option for showing or hiding news bar

### PR DESCRIPTION
**Pull request by JDurman**
Issue #1387

### CODE REVIEW

File: Hearthstone Deck Tracker/Config.cs [hunk 1]
```diff
@@ -396,7 +396,10 @@ public class Config
 		[DefaultValue(-1)]
 		public int IgnoreNewsId = -1;
 
-		[DefaultValue(true)]
+        [DefaultValue(true)]
+        public bool ShowNewsBar = true;
```
Comment (Epix37) [inline Hearthstone Deck Tracker/Config.cs]: There is no need for a new config value, checking `IgnoreNewsId != -1` is sufficient.

File: Hearthstone Deck Tracker/Core.cs [hunk 2]
```diff
@@ -118,8 +118,10 @@ public static void Initialize()
 			MainWindow.Options.OptionsTrackerPlugins.Load();
 			PluginManager.Instance.StartUpdateAsync();
 
-			UpdateOverlayAsync();
-			NewsUpdater.UpdateAsync();
+            if (Config.Instance.ShowNewsBar)
```
Comment (Epix37) [inline Hearthstone Deck Tracker/Core.cs]: Checking this removed the ability for me to remotely enable the news bar again, which might be useful is some cases.

File: Hearthstone Deck Tracker/FlyoutControls/Options/Tracker/TrackerSettings.xaml.cs [hunk 3]
```diff
@@ -357,5 +359,33 @@ private void CheckBoxAnalytics_OnUnchecked(object sender, RoutedEventArgs e)
...
+            if (!_initialized)
+                return;
+            Config.Instance.IgnoreNewsId = -1;
+            Config.Instance.ShowNewsBar = true;
+            Config.Save();
+            Utility.NewsUpdater.UpdateAsync();
+        }
+
+        private void CheckboxShowNewsBar_Unchecked(object sender, RoutedEventArgs e)
+        {
```
Comment (Epix37) [inline Hearthstone Deck Tracker/FlyoutControls/Options/Tracker/TrackerSettings.xaml.cs]: Extracting `MainWindow.BtnCloseNews_OnClick()` to a new method and calling it here would be much cleaner.

**Comment by Epix37** [comment]
Two more things:
- [Please use tabs instead of spaces.](https://github.com/Epix37/Hearthstone-Deck-Tracker/blob/master/CONTRIBUTING.md)
- Currently, closing the news bar via its integrated close button would not update the checkbox in options.

**Comment by JDurman** [comment]
Thanks for the feedback.  Yeah when I was doing it I wasn't sure if it should be a permanent disable independent of the ignore. I can definitely see why you want to remotely enable it so I'll just implement the IgnoreId functionality.  Sorry about the spaces my visual Studio was converting my tabs to spaces.

**Comment by JDurman** [comment]
Alright It's working now and I have rebased it onto the most recent commit.

**Comment by Epix37** [comment]
Merged with https://github.com/Epix37/Hearthstone-Deck-Tracker/commit/4733755a3a74cfac53843e40b1d9d68e58243a24

**Pull request closed without merging**